### PR TITLE
[wip] native JSON support

### PR DESCRIPTION
**Pull request by aol-nnov**
Hi!

I was i need of JSON data type power which PostgreSQL provides, so I've started working in this direction.
It seems, that there is a demand for that support already: #31, #7, so, @gebrits, @kesavkolla I invite you to collaborate in this area if you still interested.

@raymondfeng what do you think about enabling certain features in the connector (upsert support, json, whatever) based on the database version? Where in the code is the best way to test the DB version and fill a set of supported features? Should it be done automatically or set via config? Also tell me please what do you think about mods in this PR?

### CODE REVIEW

**Comment by aol-nnov** [comment]
nice articles that helped me a lot:
http://www.postgresql.org/docs/9.3/static/functions-json.html
http://schinckel.net/2014/05/25/querying-json-in-postgres/

File: lib/postgresql.js [hunk 1]
```diff
@@ -477,10 +477,7 @@ PostgreSQL.prototype.toDatabase = function (prop, val) {
     }
   }
   if (val.constructor.name === 'Object') {
-    if (prop.postgresql && prop.postgresql.dataType === 'json') {
```
Comment (kesavkolla) [inline lib/postgresql.js]: PostgreSQL 9.4 introduced new type jsonb.  So add jsonb also in here.

**Comment by kesavkolla** [comment]
Actually my example is bad.  The escapeName doesn't even do anything with select columns.  The escapeName is only called from createTable and columns to be dropped.  So there is no need to touch that code in my opinion.

**Comment by jmls** [comment]
did this stall here ? It would be useful to have the 9.4 json datatypes in place now

**Comment by aol-nnov** [comment]
@jmls sad to say, I've dropped it completely. Moved away from loopback and using bookshelf.js currently.

**Comment by jmls** [comment]
@aol-nnov : That's not good to hear. May I ask why ?

**Comment by aol-nnov** [comment]
@jmls well, it was a long way to go :)
first of all, I had to wrap my head around all the guts loopback have. And the best way was to start something similar from scratch, lol.
secondly, I needed better postgresql support, and, to be frank, I really love bookshelf and knex.
and my app is not only api, but a web frontend too (well, it actually does not cound as an argument and I could build web frontend atop of loopback)

**Comment by ZepAviator** [comment]
Having JSON / JSONb support would be great. It seems that maybe some ideas for code on these properties types could be pulled from the nosql connectors?

Anyone have any branches of working JSON type?

**Comment by slnode** [comment]
Can one of the admins verify this patch?

**Comment by slnode** [comment]
Can one of the admins verify this patch?

**Comment by slnode** [comment]
Can one of the admins verify this patch?

**Comment by zbarbuto** [comment]
FWIW #304 was merged recently

**Pull request closed without merging**